pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -282,7 +289,13 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
                 log.debug("Found topics on namespace {}", namespaceName);
             }
             boolean hasNonSystemTopic = false;
-            for (String topic : topics) {
+            for (String topic : partitionedTopicNames) {
+                if (!pulsar().getBrokerService().isSystemTopic(TopicName.get(topic))) {
+                    hasNonSystemTopic = true;
+                    break;
+                }
+            }
+            for (String topic : topicNames) {
```

Comment:
If `hasNonSystemTopic` is `true`, it is unnecessary here, please add a check.

Reply:
Already fixed